LICENSE

```diff
@@ -0,0 +1,201 @@
+      PARTICULAR PURPOSE. You are solely responsible for determining the
+      appropriateness of using or redistributing the Work and assume any
+      risks associated with Your exercise of permissions under this License.
+
+   8. Limitation of Liability. In no event and under no legal theory,
+      whether in tort (including negligence), contract, or otherwise,
+      unless required by applicable law (such as deliberate and grossly
+      negligent acts) or agreed to in writing, shall any Contributor be
+      liable to You for damages, including any direct, indirect, special,
+      incidental, or consequential damages of any character arising as a
+      result of this License or out of the use or inability to use the
+      Work (including but not limited to damages for loss of goodwill,
+      work stoppage, computer failure or malfunction, or any and all
+      other commercial damages or losses), even if such Contributor
+      has been advised of the possibility of such damages.
+
+   9. Accepting Warranty or Additional Liability. While redistributing
+      the Work or Derivative Works thereof, You may choose to offer,
+      and charge a fee for, acceptance of support, warranty, indemnity,
+      or other liability obligations and/or rights consistent with this
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
+
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "{}"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright 2020 NVIDIA Corporation
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
```

Comment:
In libcu++ we are also adding the LLVM exception. Should we also do that here?

https://github.com/NVIDIA/libcudacxx/blob/c6d29a7c4f7e9db51679e2a2a6f801d7ba00af79/LICENSE.TXT#L208-L234

Reply:
Our license file in the monorepo will be delightfully complex. I think this would be the correct way to structure things. 

```
==============================================================================
libcu++ is under the Apache License v2.0 with LLVM Exceptions:
==============================================================================
...

==============================================================================
Legacy LLVM License (https://llvm.org/docs/DeveloperPolicy.html#legacy):
==============================================================================
...

==============================================================================
Some libcudacxx components are covered by the below license. Each source file
indicates which license it is under.
==============================================================================
...

==============================================================================
Thrust's source code is released under the Apache License, Version 2.0:
==============================================================================
...

================================================================================
Some portions of Thrust may be licensed under other compatible open-source
licenses. Any divergence from the Apache 2 license will be noted in the source
code where applicable.
Portions under other terms include, but are not limited to:
================================================================================
<--! Boost Software License -->

==============================================================================
Portions of the thrust::complex implementation are derived from FreeBSD with the
following terms:
==============================================================================
...

==============================================================================
CUB's source code is released under the BSD 3-Clause license:
==============================================================================
...

Reply:
ACK - will update.

Reply:
I've followed this for the most part - I consolidated Thrust and libcu++ Apache 2.0 info so we didn't have to repeat the license 2x in the file.

Do we need to include https://github.com/NVIDIA/libcudacxx/blob/main/libcxx/CREDITS.TXT here?

Reply:
NAL, but I don't _think_ so? We didn't have the credits file in libcu++'s license, so I think it is fine.